### ISSUE TITLE
Extension not allowing proper engine shutdown on Windows 7

### ISSUE THREAD

**Issue by subsoap** [subsoap/discordrich#3]
We have Win7 users reporting our game is not shutting down properly when it closes, apparently it is discordrich extension that is causing it?

> Ok, we managed to find a Win7 VM to install steam and try on.
> And we could reproduce the same behavior.
> 
> By checking the option “non intrusive” (or something), we managed WinDbg to attach to the process, and this is the callstack we got.
> 
> So, next I tried bundling without the discord extension, and then the executable shut down as it should.
> 
> Regarding the callstack, we also found this hotfix which seem to describe the problem:

https://support.microsoft.com/en-us/help/2582203/a-process-that-is-being-terminated-stops-responding-in-windows-7-or-in

![screen shot 2019-02-25 at 10 15 07](https://user-images.githubusercontent.com/409170/53335731-1db0be80-38b1-11e9-8916-1a8eec2e0c96.png)

**Comment by dapetcu21** [comment]
Huh... I think we should report this upstream to [discordapp/discord-rpc](https://github.com/discordapp/discord-rpc)? @subsoap can you do it, since you seem to have more details about the issue?

**Comment by dapetcu21** [comment]
Does this still happen on Windows 7 with all Windows updates to-date? This looks like a Windows bug. I'd be suprised if they released a hotfix, but didn't fix it in a subsequent update.

**Comment by subsoap** [comment]
We have had 3 Windows 7 users report this issue to us so far. I'll ask one of them if they are fully updated.

Also Defold team checked this out and they were able to reproduce it on a Win7 VM which probably was up to date.

**Comment by subsoap** [comment]
I wonder if detecting Win7 and not allowing presence to run would be good enough until it can be fixed, because letting exe run still after user closes it is really bad.

**Comment by dapetcu21** [comment]
I'll investigate the issue starting Thursday when I get into office

**Comment by subsoap** [comment]
I tested having Win7 users run a build with and without a manual discordrich.shutdown() but didn't have an impact (fully closing was still inconsistent).

**Comment by subsoap** [comment]
A user reported game is closing now properly. Only change that maybe matters is that the Game Analytics extension was modified, not sure what else could have an impact.

Probably should try to get a hold of a Win7 VM ourselves to test on...

**Comment by subsoap** [comment]
Still getting more user reports about this, all Win7.

**Comment by subsoap** [comment]
More reports all Win7, no other OS.

**Comment by dapetcu21** [comment]
Huh... One of our colleagues has Win7 at home and doesn’t have the issue. Maybe software updates are the fix? Still didn’t get the proper time for this

**Comment by subsoap** [comment]
It is inconsistent not every time so they will need to open/close a bunch of times to see it. I sent you some extra Steam keys could you pass one of them to that Win7 user and have open close a bunch and watch processes?

**Comment by subsoap** [comment]
Got an idea on how to short circuit Win7 installs running this without it being messy?

**Comment by subsoap** [comment]
https://github.com/subsoap/discordrich/commit/adc27215102a5c4ec095855c2e12dd109525e863

Here is branch with Win7 short circuit but not sure if it's the best way. Should the value of IsWin7() be stored so it can be referenced multiple times instead of calling GetVersion() over and over again? GetVersion() is depreciated but I think it will still work for detecting Win7.

**Comment by subsoap** [comment]
Optimized version check and made it work on non-windows OS better

**Comment by subsoap** [comment]
Seems like issue is still not solved on Win7 computers, but it's not as bad. Might be another extension now at fault so will have to test and see...

**Comment by subsoap** [comment]
Yep most likely other issue is due to another extension GameAnalytics https://github.com/GameAnalytics/GA-SDK-DEFOLD/issues/11
